hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/util/ActionsValidator.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.alerts.engine.util;
+
+import static org.hawkular.alerts.api.model.trigger.TriggerAction.*;
+
+import java.util.Calendar;
+import java.util.Date;
+
+import org.hawkular.alerts.api.model.event.Alert;
+import org.hawkular.alerts.api.model.event.Event;
+import org.hawkular.alerts.api.model.trigger.TriggerAction;
+import org.hawkular.alerts.engine.log.MsgLogger;
+import org.jboss.logging.Logger;
+
+/**
+ * A Trigger can define a list of Actions that will be executed on Event/Alert generation.
+ * A Trigger can optionally define constraints based on Alert's state and/or time intervals that indicates when an
+ * action should be generated.
+ *
+ * This class is a helper to validate if an Event/Alert should be sent to the ActionsService based on the constraints
+ * defined of a specific TriggerAction object.
+ *
+ * @author Jay Shaughnessy
+ * @author Lucas Ponce
+ */
+public class ActionsValidator {
+    private static final MsgLogger msgLog = MsgLogger.LOGGER;
+    private static final Logger log = Logger.getLogger(ActionsValidator.class);
+    /**
+     * Validate if an Event should generate an Action based on the constraints defined on a TriggerAction.
+     *
+     * @param triggerAction a TriggerAction where status and time constraints are defined.
+     * @param event a given Event to validate against a TriggerAction
+     * @return true if the Event is validated and it should generated an action
+     *         false on the contrary
+     */
+    public static boolean validate(TriggerAction triggerAction, Event event) {
```

Comment:
Well, I thought as well, but I discarded as we don't need to use a programmer and to introduce that dependency just to use the Quartz format is not worth (from a tradeoff perspective).

Also the validate() method is not only related to time, as additional constraints can be added in the future, validate represents that a specific Event should generate an Action given a TriggerAction.
Additional constraints (time or not related) could be placed here and use this.

I admit that looking how Action objects are generated, it could have more sense to have a factory approach, given an Event/Alert and a TriggerAction it should generate the Action. Today this is done in one side calling just the constructor, but perhaps it makes sense to move then to this class as well.